drivers/sensor/bmi270/bmi270.c

```diff
@@ -0,0 +1,714 @@
+/*
+ * Copyright (c) 2020 Bosch Sensortec GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT bosch_bmi270
+
+#include <drivers/i2c.h>
+#include <init.h>
+#include <drivers/sensor.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+
+#include "bmi270.h"
+#include "bmi270_config_file.h"
+
+LOG_MODULE_REGISTER(bmi270, CONFIG_SENSOR_LOG_LEVEL);
+
+#define MSB_TO_DATA(lsb, msb) (int16_t)(((uint16_t) msb << 8) + (uint16_t)lsb)
+
+#ifndef BMI270_WR_LEN
+#define BMI270_WR_LEN 256
+#endif
+
+static int reg_read(uint8_t reg, uint8_t *data, uint16_t length,
+		    struct bmi270_data *dev)
+{
+	int ret = 0;
+
+	if ((dev->i2c != NULL) && (data != NULL)) {
```

Comment:
looks like `dev->i2c` will never be null if initialization took care of doing proper error handling, so this function should be simplified or entirely removed in favor of just `i2c_burst_read`